BusBoard.ConsoleApp/TflApi.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using RestSharp;
+using RestSharp.Authenticators;
+
+namespace BusBoard.ConsoleApp
+{
+    class TflApi
+    {
+        private const string BaseUrl = "https://api.tfl.gov.uk/";
+
+        readonly IRestClient client;
+
+        public TflApi()
+        {
+            client = new RestClient(BaseUrl);
+        }
+
+        public T Execute<T>(RestRequest request) where T : new()
+        {
+            var response = client.Execute<T>(request);
+
+            if (response.ErrorException != null)
+            {
+                const string message = "Error retrieving response.  Check inner details for more info.";
+                var tflException = new Exception(message, response.ErrorException);
```

Comment:
It's preferrable not to use a generic `Exception` as it doesn't tell you much about what the problem is. I would either create a custom exception to use, or just use the error exception you already have. In this case I'd probably go with the exception you already have, I don't see a reason why not to. That way you don't lose the stack trace 🙂

Reply:
That was based on the RestSharp library's template that I didn't look at carefully enough; I hadn't spotted that `response.ErrorException` was an `Exception` subclass and could be thrown. Thanks! Sorted in f72fd65